Packs/MongoDB/Integrations/MongoDB/MongoDB.py

```diff
@@ -180,13 +180,23 @@ def convert_str_to_datetime(entries: dict) -> dict:
     """
     # regex for finding timestamp in string
     regex_for_timestamp = re.compile(r'\d{4}-[01]\d-[0-3]\dT[0-2]\d:[0-5]\d:[0-5]\d(?:\.\d+)?Z?')
-    for key, value in entries.items():
-        for match in regex_for_timestamp.findall(str(value)):
+    time_format = '%Y-%m-%dT%H:%M:%S.000Z'
+    if isinstance(entries, str):
+        matches = regex_for_timestamp.findall(entries)
+        if len(matches) == 1:
             try:
-                entries[key] = datetime.strptime(match, '%Y-%m-%dT%H:%M:%S.000Z')
-            except ValueError as e:
-                raise DemistoException(str(e))
-
+                return datetime.strptime(matches[0], time_format)
+            except ValueError:
+                # If could not parse the date, it may not a date. Return entries
```

Comment:
i think this is not handled appropriately
in what cases will we get here and what will be returned?

Reply:
I don't think that it could ever get to the except, it's there for ensuring that it'll never get there.
If somehow, someone could pass a string that is both fitting the regex and not fitting the time format, it'll return the string itself.